### ISSUE TITLE
Still missing versioning

### ISSUE THREAD

**Issue by mutec** [WoltLab/d.ts#1]
The correct matching version is still missing. Currently this repo is just waste of time trying to guess what commit might be equal to which version of what package.
Why don't you use the common ways to handle such projects?

**Comment by mutec** [comment]
To illustrate how easy and comfortable it could be:

https://www.npmjs.com/package/@mysterycode/de.mysterycode.mcps.test
![image](https://user-images.githubusercontent.com/1749344/234406487-d930aca7-8390-45c5-874c-0ac4969398f4.png)

**Comment by dtdesign** [comment]
Once a new version has been released, the public API of the branch remains stable. You can simply reference the tip of that branch for your project and you’re good to go. We don’t care for the older versions within a branch and neither should you, thus any versioning that goes beyond the major/minor part is worthless. You already get this part through the distinct branches.

Publishing it to npm adds more complexity without adding any actual value apart from a touch more comfort during the initial setup.


> Currently this repo is just waste of time […]

Hopefully at some point in your life you will learn how to voice criticism without insulting people.

**Comment by mutec** [comment]
Waiting for the stable release before starting the changes would add a huge delay for upgrades after your official release. I don't think anyone would want this.
Nevertheless it's not sure that there aren't any changes in the code. I'm sure you're giving your best to avoid changes in the API itself, but shit happens.
As far as I know, requiring branches is tricky. If I remember correctly, I used to use the master-branch of the WSC, but when updating the npm stuff, it didn't update that requirement's code. I'm pretty sure I'm one of the first developers to use this repo instead of the WSC.


> without insulting people.

I didn't insult people. I just told you/the public that I spent a lot of time finding out which commit might suite for the version I was requiring in my packages which is not necessary in case it's versioned. At the moment of writing the initial message it was recommended to use this repo instead of the one of the WSC (I don't remember who recommended migrating to the d.ts-stuff, doesn't really matter for me since that's not personal but technical). No one explained whether there will be tags or something else making it possible to match the exact version of the WSC or even that it's only WSC 6.0+, not also 5.5. Last but not least I showed you a simple possibility for versioning the projects and make it possible to track changes (e.g. by using Dependabot) without the need to do anything manually since it can be done automatically during the release via GitHub Actions. Another benefit would be that it could be released independently per package, matching the explicit version of that package. At the moment it seams like it's only WSC and WBB, but not Filebase, Calendar, Gallery or other packages (maybe they don't contain any TypeScript-based code anyways - I don't have access to them, so I don't know about that).
